### DISCUSSION TITLE
Instant anchoring with multiple independent models

### DISCUSSION THREAD

**Discussion by m-paunovic-vegait** [sceneview/sceneview#104] [Q&A]
Hi!

What would be the best way to have several models instantly anchor around a common point without overlapping (i.e. arranged in a line or a circle or whatever), but still be able to be independently moved around and rotated? I've tried several things with centerOrigin and onAnchorChanged, but all my models either bunch up in the center, or behave strangely when moved.

Thanks in advance!

edit: I seem to be making some progress with the CursorNode and messing with the worldPosition of the newly-loaded models, but if I spawn them in on the first time the cursor's onPoseChanged fires, the models appear in the distance. Is there something like a confidence rating for the cursor's position that I can use, or do I just add a timeout and hope for the best?

**Comment by grassydragon** [comment]
Hi!
I think that you can either use a parent node while placing the models or dynamically calculate and update their positions from a central point that is determined by a hit result. Once the models are placed you can extract them from the parent node and anchor to their current positions.

**Comment by ThomasGorisse** [comment]
>edit: I seem to be making some progress with the CursorNode and messing with the worldPosition of the newly-loaded models, but if I spawn them in on the first time the cursor's onPoseChanged fires, the models appear in the distance. Is there something like a confidence rating for the cursor's position that I can use, or do I just add a timeout and hope for the best?

By default the `ArModelNode` (so the `CursorNode`) is using the `PlacementMode.BEST_AVAILABLE` defined with `instantPlacementFallback = true`
https://github.com/SceneView/sceneview-android/blob/6260bf716f43d7359dc667673fc1ded66c991234/arsceneview/src/main/java/io/github/sceneview/ar/node/ArNode.kt#L503-L509
And as mentioned here
https://github.com/SceneView/sceneview-android/blob/6260bf716f43d7359dc667673fc1ded66c991234/arsceneview/src/main/java/io/github/sceneview/ar/node/ArNode.kt#L443-L444
the `ArNode` will be first placed using instant placement which is not meant to be used for accuracy.
You can disable the instant placement fallback:
```kotlin
arNode.placementMode.instantPlacementFallback = false
```
or listen for `CursorNode.onHitResult` and manage if the model should be visible or not:
```kotlin
cursorNode.onHitResult = { _, hitResult ->
    val isAccurate = hitResult?.isTracking == true &&
            hitResult.trackable !is InstantPlacementPoint
    // Set the CursorNode color to red if not accurate
    cursorNode.isTracking = isAccurate
    // Hide or show the model
    modelNode.isVisible = isAccurate
}
```
Have a look at this part:
https://github.com/SceneView/sceneview-android/blob/a6e4bd39c39b8c3039e12bec5c948b360f298677/arsceneview/src/main/java/io/github/sceneview/ar/arcore/ArFrame.kt#L268-L284
